src/platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.cpp

```diff
@@ -0,0 +1,133 @@
+#include <platform/silabs/multi-ota/OTAMultiImageProcessorImpl.h>
+#include <platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.h>
+
+#include <app/clusters/ota-requestor/OTARequestorInterface.h>
+#include <platform/silabs/SilabsConfig.h>
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include "sl_si91x_driver.h"
+#ifdef __cplusplus
+}
+#endif
+
+namespace chip {
+
+CHIP_ERROR OTAWiFiFirmwareProcessor::ProcessInternal(ByteSpan & block)
+{
+    int32_t status = SL_STATUS_OK;
+    // Store the header of the OTA file
+    static uint8_t writeBuffer[kAlignmentBytes] __attribute__((aligned(4))) = { 0 };
+    // Used to tranfer other block to processor
+    static uint8_t writeDataBuffer[kBlockSize] __attribute__((aligned(4))) = { 0 };
+
+    ChipLogProgress(SoftwareUpdate, "ProcessInternal WiFi Block processing");
+    if (!mDescriptorProcessed)
+    {
+        ReturnErrorOnFailure(ProcessDescriptor(block));
+#if SL_MATTER_ENABLE_OTA_ENCRYPTION
+        /* 16 bytes to used to store undecrypted data because of unalignment */
+        mAccumulator.Init(requestedOtaMaxBlockSize + 16);
+#endif // SL_MATTER_ENABLE_OTA_ENCRYPTION
+    }
+
+#if SL_MATTER_ENABLE_OTA_ENCRYPTION
+    MutableByteSpan mBlock = MutableByteSpan(mAccumulator.data(), mAccumulator.GetThreshold());
+    memcpy(&mBlock[0], &mBlock[requestedOtaMaxBlockSize], mUnalignmentNum);
+    memcpy(&mBlock[mUnalignmentNum], block.data(), block.size());
+
+    if (mUnalignmentNum + block.size() < requestedOtaMaxBlockSize)
+    {
+        uint32_t mAlignmentNum = (mUnalignmentNum + block.size()) / 16;
+        mAlignmentNum          = mAlignmentNum * 16;
+        mUnalignmentNum        = (mUnalignmentNum + block.size()) % 16;
+        memcpy(&mBlock[requestedOtaMaxBlockSize], &mBlock[mAlignmentNum], mUnalignmentNum);
+        mBlock.reduce_size(mAlignmentNum);
+    }
+    else
+    {
+        mUnalignmentNum = mUnalignmentNum + block.size() - requestedOtaMaxBlockSize;
+        mBlock.reduce_size(requestedOtaMaxBlockSize);
+    }
+
+    OTATlvProcessor::vOtaProcessInternalEncryption(mBlock);
+    block = mBlock;
+#endif // SL_MATTER_ENABLE_OTA_ENCRYPTION
+
+    if (mFWchunktype == SL_FWUP_RPS_HEADER)
+    {
+        memcpy(&writeBuffer, block.data(), kAlignmentBytes);
+        // Send RPS header which is received as first chunk
+        status       = sl_si91x_fwup_start(writeBuffer);
+        status       = sl_si91x_fwup_load(writeBuffer, kAlignmentBytes);
+        mFWchunktype = SL_FWUP_RPS_CONTENT;
+        memcpy(&writeDataBuffer, block.data() + kAlignmentBytes, (block.size() - kAlignmentBytes));
+        status = sl_si91x_fwup_load(writeDataBuffer, (block.size() - kAlignmentBytes));
+    }
+    else if (mFWchunktype == SL_FWUP_RPS_CONTENT)
+    {
+        memcpy(&writeDataBuffer, block.data(), block.size());
+        // Send RPS content
+        status = sl_si91x_fwup_load(writeDataBuffer, block.size());
+        if (status != SL_STATUS_OK)
```

Comment:
this can be simplified to this, can be put in a verify and return as well
```cpp
if (status != SL_STATUS_OK && status != SL_STATUS_SI91X_FW_UPDATE_DONE )
{
    ChipLogError(SoftwareUpdate, "ERROR: In HandleProcessBlock sl_si91x_fwup_load() error %ld", status);
                return CHIP_ERROR_CANCELLED;
}

if (status == SL_STATUS_SI91X_FW_UPDATE_DONE)
{
    mReset = true;
 }
```

Reply:
Added `VerifyOrReturnError` and moved setting mReset to `ApplyAction` .